packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkImplementation.kt

```diff
@@ -156,6 +162,26 @@ class DdSdkImplementation(
         }
     }
 
+    private fun configureSynthetics() {
+        if (DdSdkSynthetics.testId.isNullOrBlank() || DdSdkSynthetics.resultId.isNullOrBlank()) {
+            return
+        }
+
+        // Workaround for `AdvancedRumMonitor` and `setSyntheticsAttribute` not being visible.
+        val intent = Intent().apply {
+            putExtra("_dd.synthetics.test_id", DdSdkSynthetics.testId)
+            putExtra("_dd.synthetics.result_id", DdSdkSynthetics.resultId)
+        }
+
+        val proxyActivity = Activity()
+        proxyActivity.intent = intent
+
+        val lifecycleTracking = object : ActivityLifecycleTrackingStrategy() {}
+        val core = DatadogSDKWrapperStorage.getSdkCore()
+        lifecycleTracking.register(core, appContext)
+        lifecycleTracking.onActivityCreated(proxyActivity, null)
```

Comment:
I'm not quite clear why we are doing this. Synthetics starts the application with intent having necessary attributes and we read them in `DdSdk`. And then, with any view tracking strategy registered we should be able to read it.

* read in Android SDK https://github.com/DataDog/dd-sdk-android/blob/1ce4aa77e4240fb8b7207f1672e429c2e452de1a/features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/tracking/ActivityLifecycleTrackingStrategy.kt#L95-L106
* view tracking setup in ReactNative SDK https://github.com/DataDog/dd-sdk-reactnative/blob/6d7b1a947254a6854bd8bd8672b97d71a1e0fc26/packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkNativeInitialization.kt#L123-L128

The only reason why it wouldn't be read is having `configuration.nativeViewTracking != true`. Is this a "hack" for that case?

Reply:
The problem is that you are supposed to be calling `useViewTrackingStrategy` in MainApplication, while on React Native, by the time the initialization is called, the MainActivity has already been created.

Reply:
I would rather then try to do changed in API of Android SDK, so that injection is available from cross-platform

Reply:
It might be difficult to achieve, as the method would have to be available before the initialisation of the SDK. 

However, it might be good to have a method that we can call when we register to the lifecycle on DdSdk init, to which we can pass the MainActivity intent so that the Android SDK can internally handle the synthetics attributes setting, or any other operation that has to be performed on the MainActivity.

In my opinion, for the purpose of this PR, it wouldn't be too bad to expose either the `setSyntheticsAttribute` method or the `AdvancedRumMonitor` class, so that we can at least make sure that the session type is reported correctly for synthetics.

However, I am open to other ideas, and if we manage to make the View Tracking Strategies work on the MainActivity on RN, I would still expose a basic `ViewTrackingStrategy` that does not do anything aside from consuming the intent extras for synthetics, and replace the current default `NoOpViewTrackingStrategy`.